drivers/adc/ad7124/ad7124.h

```diff
@@ -257,19 +265,139 @@
+#define AD7124_AIN_BUF_MSK                  NO_OS_GENMASK(6,5)
+
 /******************************************************************************/
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
+/**
+ * @enum	ad7124_device_type
+ * @details AD7124 Device definitions
+**/
+enum ad7124_device_type {
+	ID_AD7124_4,
+	ID_AD7124_8
+};
+
+/**
+ * @enum  ad7124_mode
+ * @brief ADC Modes of Operation
+**/
+enum ad7124_mode {
+	AD7124_CONTINUOUS,
+	AD7124_SINGLE,
+	AD7124_STANDBY,
+	AD7124_POWER_DOWN,
+	AD7124_IDLE,
+	AD7124_IN_ZERO_SCALE_OFF,
+	AD7124_IN_FULL_SCALE_GAIN,
+	AD7124_SYS_ZERO_SCALE_OFF,
+	AD7124_SYS_ZERO_SCALE_GAIN,
+	ADC_MAX_MODES
+};
+
+/**
+ * @enum ad7124_analog_input
+ * @brief ADC input sources for each channel.
+ **/
+enum ad7124_analog_input {
+	AD7124_AIN0         = 0x0,
+	AD7124_AIN1         = 0x1,
+	AD7124_AIN2         = 0x2,
+	AD7124_AIN3         = 0x3,
+	AD7124_AIN4         = 0x4,
+	AD7124_AIN5         = 0x5,
+	AD7124_AIN6         = 0x6,
+	AD7124_AIN7         = 0x7,
+	AD7124_AIN8         = 0x8,
+	AD7124_AIN9         = 0x9,
+	AD7124_AIN10        = 0xA,
+	AD7124_AIN11        = 0xB,
+	AD7124_AIN12        = 0xC,
+	AD7124_AIN13        = 0xD,
+	AD7124_AIN14        = 0xE,
+	AD7124_AIN15        = 0xF,
+	AD7124_AVSS         = 0x11,
+	AD7124_TEMP_SENSOR  = 0x10,
+	AD7124_IN_REF       = 0x12,
+	AD7124_DGND         = 0x13,
+	AD7124_AVDD_AVSS_P  = 0x14,
+	AD7124_AVDD_AVSS_M  = 0x15,
+	AD7124_IOVDD_DGND_P = 0x16,
+	AD7124_IOVDD_DGND_M = 0x17,
+	AD7124_ALDO_AVSS_P  = 0x18,
+	AD7124_ALDO_AVSS_M  = 0x19,
+	AD7124_DLDO_DGND_P  = 0x1A,
+	AD7124_DLDO_DGND_M  = 0x1B,
+	AD7124_V_20MV_P     = 0x1C,
+	AD7124_V_20MV_M     = 0x1D
+};
+
+/**
+ * @struct ad7124_analog_inputs
+ * @brief Positive and negative analog inputs
+**/
+struct ad7124_analog_inputs {
+	enum ad7124_analog_input ainp;
+	enum ad7124_analog_input ainm;
+};
 
-/*! Device register info */
+/**
+ * @struct ad7124_channel_map
+ * @brief Channel mapping
+**/
+struct ad7124_channel_map {
+	bool channel_enable;
+	uint8_t setup_sel;
+	struct ad7124_analog_inputs ain;
+};
+
+/**
+ * @enum ad7124_reference_source
+ * @brief Type of ADC Reference
+**/
+enum ad7124_reference_source {
+	/* External Reference REFIN1+/-*/
+	EXTERNAL_REFIN1 = 0x0,
+	/* External Reference REFIN2+/-*/
+	EXTERNAL_REFIN2 = 0x1,
+	/* Internal 2.5V Reference */
+	INTERNAL_REF = 0x2,
+	/* AVDD - AVSS */
+	AVDD_AVSS = 0x3,
+	/* Maximum Reference Sources */
+	MAX_REF_SOURCES
+};
+
+/**
+  * @struct ad7124_channel_setup
+  * @brief Channel setup
+**/
+struct ad7124_channel_setup {
+	bool bi_unipolar;
+	bool ref_buff;
+	bool  ain_buff;
+	enum ad7124_reference_source ref_source;
+};
+
+/* Device register info */
 struct ad7124_st_reg {
 	int32_t addr;
 	int32_t value;
 	int32_t size;
 	int32_t rw;
 };
 
-/*! AD7124 registers list*/
+/* AD7124 registers list */
 enum ad7124_registers {
 	AD7124_Status = 0x00,
```

Comment:
No need to add 0x00 explicit declaration. First member will have value of 0 already.